services/app-api/src/domain-models/contractAndRates/dataValidatorHelpers.ts

```diff
@@ -93,4 +113,82 @@ const validateContractDraftRevisionInput = (
     return parsedData.data
 }
 
-export { validateContractDraftRevisionInput }
+const parseContract = (
+    contract: ContractType,
+    stateCode: string,
+    store: Store,
+    featureFlags?: FeatureFlagSettings
+): ContractType | Error => {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
+    const formData = contract.draftRevision?.formData!
+    const parsedData = contractSchema
```

Comment:
I think we need to modify this function with a new schema to validate that the contract data is all there before submitting.

`parseContract` is being used to validate contract data before submitting.`contractFormDataSchema` is the schema used for parsing contract `formData` and almost all of the fields are optional is this schema. So invalid data will be able to be submitted. Testing this with the `parseContract` - `'success if valid form data'` test, you can set fields like `contractDateStart`, `contractDateEnd`, or `contractExecutionStatus` as `undefined` and it will pass the test when it should not for submitting a contract.

The function itself is not an issue, but using this function as is in submitContract would let contracts with invalid data be submitted.